hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCacheInvalidationTest.java

```diff
@@ -0,0 +1,355 @@
+
+    static final String DEFAULT_CACHE_NAME = "com.hazelcast.client.cache.nearcache.ClientNearCacheInvalidationTest";
+
+    // time to wait until invalidation event is delivered (when used with assertTrueEventually)
+    // and time to wait when testing that no invalidation event is delivered (used with assertTrueAllTheTime)
+    static final int TIMEOUT = 5;
+
+    // when true, invoke operations which are supposed to deliver invalidation events from a Cache instance on a member
+    // otherwise use the client-side proxy.
+    @Parameter
+    public boolean invokeCacheOperationsFromMember;
+
+    @Parameter(1)
+    public InMemoryFormat inMemoryFormat;
+
+    @Parameters(name = "fromMember:{0}, format:{1}")
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(new Object[][]{
+                {false, InMemoryFormat.BINARY},
+                {false, InMemoryFormat.OBJECT},
+                {true, InMemoryFormat.BINARY},
+                {true, InMemoryFormat.OBJECT},
+        });
+    }
+
+    private HazelcastInstance member;
+    private TestHazelcastFactory hazelcastFactory;
+    private NearCacheTestContext testContext;
+
+    @Before
+    public void setup() {
+        hazelcastFactory = new TestHazelcastFactory();
+        member = hazelcastFactory.newHazelcastInstance(getConfig());
+        ClientConfig clientConfig = createClientConfig();
+        clientConfig.addNearCacheConfig(createNearCacheConfig(inMemoryFormat));
+        HazelcastClientProxy client = (HazelcastClientProxy) hazelcastFactory.newHazelcastClient(clientConfig);
+        NearCacheManager nearCacheManager = client.client.getNearCacheManager();
+        CachingProvider provider = HazelcastClientCachingProvider.createCachingProvider(client);
+        HazelcastClientCacheManager cacheManager = (HazelcastClientCacheManager) provider.getCacheManager();
+        HazelcastServerCacheManager memberCacheManager = (HazelcastServerCacheManager) HazelcastServerCachingProvider
+                .createCachingProvider(member).getCacheManager();
+
+        ICache<Object, String> cache = cacheManager.createCache(DEFAULT_CACHE_NAME, createCacheConfig(inMemoryFormat));
+        ICache<Object, String> memberCache = member.getCacheManager().getCache(
+                CacheUtil.getPrefixedCacheName(DEFAULT_CACHE_NAME, null, null));
+
+        NearCache<Data, String> nearCache = nearCacheManager.getNearCache(
+                cacheManager.getCacheNameWithPrefix(DEFAULT_CACHE_NAME));
+
+        testContext = new NearCacheTestContext(client, member, cacheManager, memberCacheManager, nearCacheManager, cache,
+                memberCache, nearCache);
+
+        // add 1 K-V pair in the cache
+        testContext.cache.put("a", "b");
+        testContext.cache.get("a");
+
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.shutdownAll();
+    }
+
+    @Test
+    public void when_cacheDestroyed_invalidationEventIsReceived() {
```

Comment:
Sounds good, will incorporate this assertion in test cases where an invalidation event is expected to be received.